### PR TITLE
Add conversion for instance(), set_shader_param() and node path literals

### PR DESCRIPTION
**Pull request by and3rson**
This addresses some items from #64003:

- `.instance(.*)` -> `.instantiate($1)` (guarded against messing up the .tscn metadata)
- Improved `get_object_of_execution` to handle node path literals:
    ```gd
    # 3.x code:
    var velocity = $viewport_container/viewport/ball.move_and_slide(velocity, Vector3.UP)
    # 4.x before:
    ball.set_motion_velocity(velocity)
    ball.set_up_direction(Vector3.UP)
    ball.move_and_slide()
    velocity = $viewport_container/viewport/ball.velocity  # <- this line is correct
    # 4.x with this PR:
    $viewport_container/viewport/ball.set_velocity(velocity)
    $viewport_container/viewport/ball.set_up_direction(Vector3.UP)
    $viewport_container/viewport/ball.move_and_slide()
    velocity = $viewport_container/viewport/ball.velocity
    ```
- `set_shader_param` -> `set_shader_uniform`

### CODE REVIEW

**Comment by and3rson** [comment]
@akien-mga wow, you're really fast! :D

File: editor/project_converter_3_to_4.cpp [hunk 1]
```diff
@@ -516,6 +516,7 @@ static const char *gdscript_function_renames[][2] = {
 	{ "set_region_filter_clip", "set_region_filter_clip_enabled" }, // Sprite2D
 	{ "set_rotate", "set_rotates" }, // PathFollow2D
 	{ "set_scancode", "set_keycode" }, // InputEventKey
+	{ "set_shader_param", "set_shader_uniform" }, // InputEventWithModifiers
```
Comment (fire-forge) [inline editor/project_converter_3_to_4.cpp:519]:
```suggestion
	{ "set_shader_param", "set_shader_uniform" }, // ShaderMaterial
```

File: editor/project_converter_3_to_4.cpp [hunk 2]
```diff
@@ -2188,6 +2188,15 @@ bool ProjectConverter3To4::test_conversion(const RegExContainer &reg_container)
 	}
 	valid = valid & (got == expected);
 }
+{
+	String base = "var node = $world/ukraine/lviv.";
+	String expected = "$world/ukraine/lviv.";
```
Comment (fire-forge) [inline editor/project_converter_3_to_4.cpp:2193]: What is this for?

Comment (and3rson) [reply editor/project_converter_3_to_4.cpp:2193]: This is a test case that validates if `get_object_of_execution` properly handles node path literals.

**Review by qarmin — APPROVED** [review]

**Comment by clayjohn** [comment]
Looks great! Thanks!

**Pull request merged**